pkg/volume/flexvolume/attacher-defaults.go

```diff
@@ -37,13 +39,15 @@ func (a *attacherDefaults) WaitForAttach(spec *volume.Spec, timeout time.Duratio
 }
 
 // GetDeviceMountPath is part of the volume.Attacher interface
-func (a *attacherDefaults) GetDeviceMountPath(spec *volume.Spec) string {
-	driverDir := path.Join(a.host.GetPluginDir(flexVolumePluginName), a.driverName)
-
-	if spec.PersistentVolume != nil {
-		return path.Join(driverDir, "pvs", spec.PersistentVolume.Name)
+func (a *attacherDefaults) GetDeviceMountPath(spec *volume.Spec, mountsDir string) string {
+	deviceName, err := a.plugin.GetDeviceName(spec)
+	if err != nil {
+		glog.Errorf("Unable to call GetDeviceName from GetDeviceMountPath: %s", err)
+		return ""
 	}
-	return path.Join(driverDir, "pods", string(spec.PodUID), spec.Volume.Name)
+
+	// deviceName must be in /dev, and we strip this prefix
+	return path.Join(mountsDir, deviceName[len("/dev/"):])
```

Comment:
Thinking more - GetDeviceName need not return starting with "/dev". It can just return device name, which can be globally unique, like pd_name.